### DISCUSSION TITLE
Enhancement ideas

### DISCUSSION THREAD

**Discussion by kaisalmen** [TypeFox/monaco-languageclient#773] [Ideas]
Let's gather ideas for future improvements improvements:
- `monaco-editor-wrapper`: Extend the Wrapper to support views and workbench service. Currently it only support the editor service. You can do it all manually, but it does make sense to ease the usage via helper code and typed configuration
  - [x] Views service integration
  - [ ] Workbench service integration. Preparation work already in place with v6
- Review and enhance vscode-ws-jsonrpc  (#748)
- Taken from https://github.com/TypeFox/monaco-languageclient/issues/595: Do we want to integrate similar functionality from [languageserver-mutualized](https://github.com/CodinGame/languageserver-mutualized) here?
- Rethink examples handling. Move them out of the npm workspace and packages directory into a separate self-standing package directory (e.g. examples)
- Add a new example / change an existing example that integrates the Debug Adapter Protocol. (see exists already #690)
- Is it possible to further decrease the bundle size (see #678)
- We should think about further improvements to the documentation and extend the code doc of the examples.
- Providing a way to synchronize files between the client and the server easily, https://github.com/TypeFox/monaco-languageclient/issues/834

We can create further issues once we identified good candidates and volunteers to implement them.

**Comment by kaisalmen** [comment]
@CGNonofr I went through the issues and closed 10+ and extracted some enhancement ideas. They are not prioritized. If you have further things to add, please do so.

**Reply by CGNonofr** [reply]
What about providing a way to synchronize files between the client and the server easily?

**Reply by kaisalmen** [reply]
Good point. That's the work I already started.

**Reply by kaisalmen** [reply]
I created https://github.com/TypeFox/monaco-languageclient/issues/834 to track it.
